arviz/plots/ecdfplot.py

```diff
@@ -180,75 +181,47 @@ def plot_ecdf(
     values = np.ravel(values)
     values.sort()
 
-    ## This block computes gamma and uses it to get the upper and lower confidence bands
-    ## Here we check if we want confidence bands or not
-    if confidence_bands:
-        ## If plotting PIT then we find the PIT values of sample.
-        ## Basically here we generate the evaluation points(x) and find the PIT values.
-        ## z is the evaluation point for our uniform distribution in compute_gamma()
-        if pit:
-            x = np.linspace(1 / npoints, 1, npoints)
-            z = x
-            ## Finding PIT for our sample
-            probs = cdf(values) if cdf else compute_ecdf(values2, values) / len(values2)
-        else:
-            ## If not PIT use sample for plots and for evaluation points(x) use equally spaced
-            ## points between minimum and maximum of sample
-            ## For z we have used cdf(x)
-            x = np.linspace(values[0], values[-1], npoints)
-            z = cdf(x) if cdf else compute_ecdf(values2, x)
-            probs = values
-
-        n = len(values)  # number of samples
-        ## Computing gamma
-        gamma = fpr if pointwise else compute_gamma(n, z, npoints, num_trials, fpr)
-        ## Using gamma to get the confidence intervals
-        lower, higher = get_lims(gamma, n, z)
-
-        ## This block is for whether to plot ECDF or ECDF-difference
-        if not difference:
-            ## We store the coordinates of our ecdf in x_coord, y_coord
-            x_coord, y_coord = get_ecdf_points(x, probs, difference)
+    if pit:
+        eval_points = np.linspace(1 / npoints, 1, npoints)
+        if cdf:
+            sample = np.apply_along_axis(cdf, 0, values)
```

Comment:
I take the apply_along_axis is so `cdf` can take scalar input right? Given the common case I can think of is using the `.cdf` method of scipy distributions, it should be documented, otherwise it might not be used at all. If we forced `cdf` to take array inputs, it should be equivalent to `cdf(values)` as we have already made sure `values` is a 1-dimensional array

Reply:
Yes, correct. The existing code already assumes `cdf` is a scalar function, and the docstring did not state otherwise, so it could be breaking to make `cdf` take a vector. Since this PR is not trying to change existing functionality, I've left it as is. But a future PR will change `cdf` to take array values and document this.

Reply:
Aaah, okay, I misremembered. It seems it assumes both scalars and arrays are acceptable, e.g. https://github.com/arviz-devs/arviz/blob/0a91fafca89237af681980aa994576873ef03283/arviz/plots/ecdfplot.py#L215-L218. Since we'll only ever need `cdf` to take array values, I'll document this expectation and update the code accordingly.

Reply:
Done!